src/application/view_photo.c

```diff
@@ -0,0 +1,465 @@
+            .bgExtPltt = GX_BG_EXTPLTT_01,
+            .priority = 0,
+            .areaOver = GX_BG_AREAOVER_XLU,
+            .mosaic = 0,
+        };
+        InitBgFromTemplate(viewPhoto->bgConfig, GF_BG_LYR_SUB_1, &bgTemplate, GF_BG_TYPE_TEXT);
+        BgClearTilemapBufferAndCommit(viewPhoto->bgConfig, GF_BG_LYR_SUB_1);
+    }
+
+    {
+        BgTemplate bgTemplate = {
+            .x = 0,
+            .y = 0,
+            .bufferSize = GF_BG_BUF_SIZE_256x256_4BPP,
+            .baseTile = 0,
+            .size = GF_BG_SCR_SIZE_256x256,
+            .colorMode = GX_BG_COLORMODE_16,
+            .screenBase = GX_BG_SCRBASE_0x7000,
+            .charBase = GX_BG_CHARBASE_0x00000,
+            .bgExtPltt = GX_BG_EXTPLTT_01,
+            .priority = 3,
+            .areaOver = GX_BG_AREAOVER_XLU,
+            .mosaic = 0,
+        };
+        InitBgFromTemplate(viewPhoto->bgConfig, GF_BG_LYR_SUB_2, &bgTemplate, GF_BG_TYPE_TEXT);
+        BgClearTilemapBufferAndCommit(viewPhoto->bgConfig, GF_BG_LYR_SUB_2);
+    }
+
+    {
+        BgTemplate bgTemplate = {
+            .x = 0,
+            .y = 0,
+            .bufferSize = GF_BG_BUF_SIZE_256x256_4BPP,
+            .baseTile = 0,
+            .size = GF_BG_SCR_SIZE_256x256,
+            .colorMode = GX_BG_COLORMODE_256,
+            .screenBase = GX_BG_SCRBASE_0x6800,
+            .charBase = GX_BG_CHARBASE_0x04000,
+            .bgExtPltt = GX_BG_EXTPLTT_01,
+            .priority = 2,
+            .areaOver = GX_BG_AREAOVER_XLU,
+            .mosaic = 0,
+        };
+        InitBgFromTemplate(viewPhoto->bgConfig, GF_BG_LYR_SUB_3, &bgTemplate, GF_BG_TYPE_256x16PLTT);
+        BgClearTilemapBufferAndCommit(viewPhoto->bgConfig, GF_BG_LYR_SUB_3);
+    }
+
+    BG_ClearCharDataRange(GF_BG_LYR_SUB_1, 0x20, 0, viewPhoto->heapId);
+    BG_ClearCharDataRange(GF_BG_LYR_SUB_3, 0x40, 0, viewPhoto->heapId);
+    BgSetPosTextAndCommit(viewPhoto->bgConfig, GF_BG_LYR_SUB_3, BG_POS_OP_SET_X, -4);
+}
+
+static void ViewPhotoSysTask_ReleaseBgLayers(ViewPhotoSysTaskData *viewPhoto) {
+    BgSetPosTextAndCommit(viewPhoto->bgConfig, GF_BG_LYR_SUB_3, BG_POS_OP_SET_X, 0);
+    FreeBgTilemapBuffer(viewPhoto->bgConfig, GF_BG_LYR_SUB_3);
+    FreeBgTilemapBuffer(viewPhoto->bgConfig, GF_BG_LYR_SUB_2);
+    FreeBgTilemapBuffer(viewPhoto->bgConfig, GF_BG_LYR_SUB_1);
+    GXS_SetGraphicsMode(GX_BGMODE_0);
+}
+
+static void ViewPhotoSysTask_LoadBgGraphics(ViewPhotoSysTaskData *viewPhoto) {
+    NARC *narc = NARC_New(NARC_a_1_7_1, viewPhoto->heapId);
+    GfGfxLoader_GXLoadPalFromOpenNarc(narc, 4, GF_PAL_LOCATION_SUB_BG, (enum GFPalSlotOffset)0, 0, viewPhoto->heapId);
+    GfGfxLoader_LoadCharDataFromOpenNarc(narc, 11, viewPhoto->bgConfig, GF_BG_LYR_SUB_2, 0, 0, FALSE, viewPhoto->heapId);
+    GfGfxLoader_LoadScrnDataFromOpenNarc(narc, 12, viewPhoto->bgConfig, GF_BG_LYR_SUB_2, 0, 0, FALSE, viewPhoto->heapId);
+
+    void *ncgrFile;
+    NNSG2dCharacterData *pCharData;
+    u8 r3;
```

Comment:
r3 is some kinda Icon ID?

Reply:
could be? it seems to be used more as a tile data offset but i'd rather not give it a name before we figure out what's actually being loaded here